stdlib/public/core/Arrays.swift.gyb

```diff
@@ -1624,23 +1598,22 @@ extension ${Self} {
     return try body(&inoutBufferPointer)
   }
 
-  @discardableResult
   public func _copyContents(
-    initializing ptr: UnsafeMutablePointer<Element>
-  ) -> UnsafeMutablePointer<Element> {
-    if let s = self._baseAddressIfContiguous {
-      let count = self.count
-      ptr.initialize(from: s, count: count)
+    initializing buf: UnsafeMutableBufferPointer<Element>
+  ) -> Iterator? {
+    // FIXME: decide if this really is a fatalError
+    guard var p = buf.baseAddress 
+      else { fatalError("Attempt to copy contents into nil buffer pointer") }
+    if let s = _baseAddressIfContiguous {
+      p.initialize(from: s, count: self.count)
       _fixLifetime(self._owner)
```

Comment:
I wish all _fixLifetime calls came with a comment. In this case, AFAICT the issue is that we need a _fixLifetime bracketing the _baseAddressIfContiguous getter and all uses of the pointer it returns.